hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -729,6 +729,7 @@ private void removeMember(MemberImpl member, String reason, boolean shouldCloseC
             }
 
             logger.info("Removing " + member);
+            // the following is only used when sync join strategy is used
```

Comment:
Nit-pick: It might be a good idea to have these functions no-op when the sync strategy isn't being used, to avoid overhead since they'll provide no benefit. Not a hot area though, so not a big concern.

Reply:
I'd prefer to leave this alone for now.